presto-hive/src/main/java/io/prestosql/plugin/hive/security/SqlStandardAccessControl.java

```diff
@@ -390,6 +392,14 @@ public void checkCanShowRoleGrants(ConnectorSecurityContext context, String cata
     {
     }
 
+    @Override
+    public void checkCanExecuteProcedure(ConnectorSecurityContext context, SchemaProcedureName procedure)
+    {
+        if (!isAdmin(context)) {
```

Comment:
I don't think this is right.  This would limit important table maintenance to only admins, and instead it should really be based on the procedure and the permissions the user has.  For example, I would expect anyone with `DELETE` permissions on a table to be able to execute `unregister_partition`.

Reply:
I think we need to allow the procedure itself to perform access control checks based on the object the action is performed on, since procedures are effectively connector-defined operations. We can do this by allowing the procedure to accept `ConnectorAccessControl` and `ConnectorSecurityContext` (just like `ConnectorSession`). The access control implementation would be a facade over the global access control (and thus would delegate to any system or connector access control applicable to the procedure's catalog).

Reply:
I agree. Also we would need to check usage of `external_location` when. Hive connector is configured in a way it disallows to use that property.

Reply:
> For example, I would expect anyone with `DELETE` permissions on a table to be able to execute `unregister_partition`.

That could work. What checks should the `register_partition` procedure do?